pkg/syncer/status/status_controller.go

```diff
@@ -84,13 +85,19 @@ func NewStatusSyncer(syncerLogger logr.Logger, syncTargetWorkspace logicalcluste
 			logger.V(2).Info("Set up downstream informer", "gvr", gvr.String())
 			return cache.ResourceEventHandlerFuncs{
 				AddFunc: func(obj interface{}) {
+					unstrob := obj.(*unstructured.Unstructured)
+					if hasDelayStatusSyncingAnnotation(unstrob) {
+						logger.V(2).Info("Delay status syncing", "gvr", gvr.String(), "name", unstrob.GetName(), "namespace", unstrob.GetNamespace(), "since annotation is set", storage.DelayStatusSyncing)
+						return
+					}
+
 					c.AddToQueue(gvr, obj, logger)
 				},
 				UpdateFunc: func(oldObj, newObj interface{}) {
 					oldUnstrob := oldObj.(*unstructured.Unstructured)
 					newUnstrob := newObj.(*unstructured.Unstructured)
 
-					if !deepEqualFinalizersAndStatus(oldUnstrob, newUnstrob) {
+					if notDeepEqualFinalizersAndStatusOrDelayStatusSyncingAnnotationChanged(oldUnstrob, newUnstrob) {
```

Comment:
Why does one function need to handle both cases?

Reply:
This way it's easier to return true if the annotation got removed and the finalizer/status did not change. So testing everything at once felt clearer.